paper/paper.md

```diff
@@ -0,0 +1,171 @@
+- name: Jonathan Frederic
+  orcid: 0000-0003-4805-2216
+  affiliation: 2
+- name: Brian Granger
+  orcid: 
+  affiliation: 3
+- name: Jessica Hamrick
+  orcid: 0000-0002-3860-0429
+  affiliation: 4
+- name: Logan Page
+  orcid: 0000-0002-5799-8524
+  affiliation: 5
+- name: Fernando Pérez
+  orcid: 
+  affiliation: 1
+- name: Benjamin Ragan-Kelley
+  orcid: 0000-0002-1023-7082
+  affiliation: 6
+- name: Jordan W. Suchow
+  orcid: 0000-0001-9848-4872
+  affiliation: 1
+- name: Carol Willing
+  orcid: 0000-0002-9817-8485
+  affiliation: 3
+affiliations:
+- name: University of California, Berkeley
+  index: 1
+- name: Google Inc.
+  index: 2
+- name: Cal Poly, San Luis Obispo
+  index: 3
+- name: DeepMind
+  index: 4
+- name: University of Pretoria
+  index: 5
+- name: Simula Research Laboratory
+  index: 6
+- name: Bryn Mawr College
+  index: 7
+date: 2 June 2018
+bibliography: paper.bib
+---
+
+*Note: Authors on this paper are listed in alphabetical order.*
+
+# Summary
+
+nbgrader is a flexible tool for creating and grading assignments in the Jupyter
+notebook [@kluyver2016jupyter]. nbgrader allows instructors to create a single,
+master copy of an assignment, including tests and canonical solutions. From the
+master copy, a student version is generated without the solutions, thus
+obviating the need to maintain two separate versions. nbgrader also
+automatically grades submitted assignments by executing the notebooks and
+storing the results of the tests in a database. After auto-grading, instructors
+can manually grade free responses and provide partial credit using the
+*formgrader* notebook extension. Finally, instructors can use nbgrader to leave
+personalized feedback for each student's submission, including comments as well
+as detailed error information.
+
+nbgrader can also be used with JupyterHub [@JupyterHub], which is a centralized,
+server-based installation that manages user logins and management of Jupyter
+notebooks. When used with JupyterHub, nbgrader provides additional workflow
+functionality, covering the entire grading process. After creating an
+assignment, instructors can distribute it to students, who can then fetch a copy
+of the assignment directly through the notebook server interface. Students can
+submit their completed assignment through the same interface, making it
+available for instructors. After students submit their assignments, instructors
+can collect the assignments with a single command and use the auto-grading
+functionality in the normal way.
+
+Since its conception in September 2014, nbgrader has been used in a
+number of educational contexts, including courses at UC Berkeley, Cal Poly,
+University of Pretoria, University of Edinburgh, Northeastern University,
+Central Connecticut State University, KTH Royal Institute of Technology, CU
+Boulder, University of Amsterdam, George Washington University, Texas A&M, Bryn
+Mawr College, and University of Maryland; and, as of May 2018, over 10,000
```

Comment:
Unless I am reading the [query](https://github.com/search?l=Jupyter+Notebook&q=nbgrader&type=Code) incorrectly, I think that this is 10,000 *cells* rather than notebooks. I'll see if there is a way to get unique matches...

Reply:
Running a program using the github search API now... should be able to give an exact count of nbgrader notebooks, and perhaps also estimate the number of classes...

Reply:
Yuck... having to get the notebook data in chunks based on... file size.

Reply:
Results:

* 10,476 notebooks listed through query (may have dups)
* 10,185 notebooks metadata got through API, with:
  * 8,433 non-checkpoint notebooks
  * 4,909 unique "names" (eg, "Sem2/Python/Assignment4/sheet4_Eugen1.ipynb", "lab3.ipynb", etc) -- collapses master/branch vs blob versions
  * 1,729 unique repos (eg, "willingc/jhubdocs", "hglanz/phys202-2015-work", etc)

I saved all of the metadata on these files, if anyone thinks of something else to look at.

Reply:
This is awesome, thanks so much @dsblank ! So I think we can safely keep the "over 10,000" bit. Do you think it's also worth emphasizing 1500+ repos?